clang/include/clang/Driver/Options.td

```diff
@@ -2876,6 +2876,15 @@ def flax_vector_conversions : Flag<["-"], "flax-vector-conversions">, Group<f_Gr
 def flimited_precision_EQ : Joined<["-"], "flimited-precision=">, Group<f_Group>;
 def fapple_link_rtlib : Flag<["-"], "fapple-link-rtlib">, Group<f_Group>,
   HelpText<"Force linking the clang builtins runtime library">;
+
+/// ClangIR-specific options - BEGIN
+def fclangir_enable : Flag<["-"], "fclangir-enable">, Visibility<[ClangOption, CC1Option]>,
+  Group<f_Group>, HelpText<"Use ClangIR pipeline to compile">,
+  MarshallingInfoFlag<FrontendOpts<"UseClangIRPipeline">>;
+def emit_cir : Flag<["-"], "emit-cir">, Visibility<[ClangOption, CC1Option]>,
```

Comment:
Should this be a cc1-only option, same as `-emit-obj` and `-emit-llvm-bc`? (It's always struck me as odd that `-emit-llvm` is not a cc1-only option...)

Reply:
My preference is to match current `-emit-llvm` behavior (i.e. be a driver level flag). It makes it convenient for us to just replace `-emit-llvm` with `-emit-cir` in workflows that currently expect LLVM output. If someone fixes `-emit-llvm` someday (which I find unlikely), we could then do the same for CIR. If this is a over my dead body argument for you, then whatever is fine, just wanna make sure I express what I believe is the right approach.

Reply:
> Should this be a cc1-only option, same as -emit-obj and -emit-llvm-bc? (It's always struck me as odd that -emit-llvm is not a cc1-only option...)

I prefer it being a driver flag as well if there's no strong protest.

Reply:
These `-emit-*` options do not match the convention of driver options. And there is a conflicting driver option `-e`.

It would be better to not add a driver `-emit-cir`.

Reply:
> These -emit-* options do not match the convention of driver options. And there is a conflicting driver option -e.

It matches `-emit-llvm`. That is the target behavior of `-emit-cir`.

Reply:
IMO, there is no good reason for `-emit-llvm` to be a driver option, other than for historical reasons.  That is, if we were putting it in today, we'd probably not expose it that way.  Because of that, I'd like to not repeat that mistake with `emit-cir`.